process/transaction/shardProcess.go

```diff
@@ -373,7 +373,8 @@ func (txProc *txProcessor) processTxFee(
 		return nil, nil, err
 	}
 
-	if !txProc.flagPenalizedTooMuchGas.IsSet() {
+	if !txProc.flagMetaProtection.IsSet() {
+		//backwards compatibility return
 		totalCost := core.SafeMul(tx.GasLimit, tx.GasPrice)
 		return moveBalanceFee, totalCost, nil
 	}
```

Comment:
there is an actual bug in the new implementation, because of which receipts will not be created. line 382 should be return moveBalanceFee, totalGiven, nil